sqlglot/dialects/spark2.py

```diff
@@ -114,6 +114,15 @@ def _unqualify_pivot_columns(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def _insert_sql(self: Hive.Generator, expression: exp.Insert) -> str:
+    if not expression.expression.args.get("with"):
+        return self.insert_sql(expression)
+    expression = expression.copy()
+    with_ = expression.expression.args.pop("with")
+    expression.set("with", with_)
+    return self.insert_sql(expression)
```

Comment:
Can clean this up a bit by inverting the condition:

```suggestion
    if expression.expression.args.get("with"):
        expression = expression.copy()
        expression.set("with", expression.expression.args.pop("with"))

    return self.insert_sql(expression)
```

Reply:
Nice!